dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -127,6 +127,15 @@ def toggle_hidden
     render json: []
   end
 
+  def instructions
+    require_levelbuilder_mode
+    authorize! :read, ScriptLevel
+
+    script = Script.get_from_cache(params[:script_id])
+
+    render 'levels/instructions', locals: { stages: script.stages }
```

Comment:
I actually like having `stages = local_assigns[:stages]` for similar reasons - this makes it explicit where I'm getting it from. Without it, I don't know where stages is coming from - i.e. maybe it's coming in via local_assigns, maybe it's a method somewhere (i.e. `current_user`). It's possible this is me my lack of Ruby expertise coming through, but one of the things I struggle with most in our Ruby codebase is understanding where variables/methods are defined, and thus I like to make things as explicit as I can.